motor/providers/aws/provider.go

```diff
@@ -45,6 +48,30 @@ func WithProfile(profile string) ProviderOption {
 	}
 }
 
+func WithExternalId(id string) func(o *stscreds.AssumeRoleOptions) {
+	if id != "" {
+		return func(o *stscreds.AssumeRoleOptions) {
+			o.ExternalID = &id
+		}
+	}
+	return func(o *stscreds.AssumeRoleOptions) {}
+}
+
+func WithAssumeRole(roleArn string, defaultCfg aws_sdk.Config, externalId string) ProviderOption {
```

Comment:
[super nit] the order of the arguments seems maybe weird. Like why is defaultCfg between roleArn and externalId